robottelo/test.py

```diff
@@ -101,6 +102,99 @@
 LOGGER = logging.getLogger(__name__)
 
 
+class _AssertNotRaisesContext(object):
+    """A context manager used to implement :meth:`TestCase.assertNotRaises`.
+    """
+
+    def __init__(self, expected, test_case, expected_regex=None,
+                 cli_return_code=None, http_status_code=None):
```

Comment:
@abalakht, as understood  we have a general usage for  expected_exception, but for callable value  we stick to only cli_status_code, http_status.

waht do you think if we change this to a value handler  for example 
```python
class APIRaisesHandler:
     __init__(self, value):
     self.value = value

    def response_code(self, exc_value):
         return (
                      True self.value == exc_value.response.status_code
                        else False
                    )

class CLIRaisesHandler:
     __init__(self, value):
     self.value = value

    def response_code(self, exc_value):
         return (
                      True self.value == exc_value.return_code
                        else False
                    )
```

```python 
 with self.assertNotRaises(HTTPError, value_handler=APIRaisesHandler(404)):
             host.read()
```
and this should lead to a more general usage, perhaps we should keep   cli_return_code=None, http_status_code=None, but intead of ceating an interface=api or = cli , add the value_handler in args and if it's not available look at the special cli_return_code and http_status_code and recreate the value handler.
and at __exit__ use the response_code of the handler 
what do you think?